### ISSUE TITLE
Allow analysts to view removed domain managers

### ISSUE THREAD

**Issue by vickyszuchin** [cisagov/manage.get.gov#1606]
### Issue description

As a CISA analyst
I want to know which domain managers have been removed
so I can know of all the domain managers that have been domain managers for this domain.

### Acceptance criteria

- [ ] CISA analysts can view removed domain managers
- [ ] A removed status is added
- [ ] if the status is "removed", then this does not allow this person to count as a domain manager (and does not see the domain on their overview page)
- [ ] if the status is "removed", then ID proofing is required if they have no active domains under their management and are not on the Verified by staff list

### Additional context
Work with the design team, as needed, to implement.

**implementation note:** This requires adding a status to UserDomainRole which means, at the same time you need to refactor any place using UserDomainRole and ensure that the filters are checking for the status "Manager". You may need to adjust unit tests as well. 
### Links to other issues

- Related to: #1484 
- Similar to: #1604

**Comment by abroddrick** [comment]
@Katherine-Osos Two ways we can do this is 1.) removed domain managers should gets it's own table below where we show them current domain managers and we hide it behind a show/hide or 2.) we show the count and the count is a link to a page that is filtered for this status and this domain on the User Domain Role table ( and follow what we do on portfolios). With #2 we don't have to worry about eventually needing to paginate if someone has a lots of former users. (Though maybe that is rarely an issue)

**Comment by katypies** [comment]
I agree that #2611 is likely to address the underlying need here, albeit in a slightly different way. Closing this one - I think we should do 2611, and iterate to make it more robust, add filters, and similar.
